plugins/TwoFactorAuth/Validator.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\TwoFactorAuth;
+
+use Piwik\Piwik;
+use Piwik\Session\SessionFingerprint;
+use Exception;
+use Piwik\SettingsPiwik;
+
+class Validator
+{
+    /**
+     * @var TwoFactorAuthentication
+     */
+    private $twoFa;
+
+    public function __construct(TwoFactorAuthentication $twoFactorAuthentication)
+    {
+        $this->twoFa = $twoFactorAuthentication;
+    }
+
+    public function canUseTwoFa()
+    {
+        if (!SettingsPiwik::isPiwikInstalled()) {
+            return false;
+        }
+
+        return !Piwik::isUserIsAnonymous();
+    }
+
+    public function checkCanUseTwoFa()
+    {
+        Piwik::checkUserIsNotAnonymous();
+    }
```

Comment:
Should this also have the `SettingsPiwik::isPiwikInstalled()` check?

Reply:
yep, makes sense 👍  will update.